cupy_backends/cuda/libs/_cnvrtc.pxi

```diff
@@ -0,0 +1,97 @@
+import sys as _sys
+
+from cupy_backends.cuda.api cimport runtime
+from cupy_backends.cuda._softlink cimport SoftLink
+
+
+cdef int _runtime_version = runtime.runtimeGetVersion()
+cdef str _prefix = 'nvrtc'
+cdef object _libname = None
+
+if CUPY_CUDA_VERSION != 0:
+    if 11020 <= _runtime_version < 12000:
+        # CUDA 11.x (11.2+)
+        if _sys.platform == 'linux':
+            _libname = 'libnvrtc.so.11.2'
+        else:
+            _libname = 'nvrtc64_112_0.dll'
+    elif 12000 <= _runtime_version < 13000:
+        # CUDA 12.x
+        if _sys.platform == 'linux':
+            _libname = 'libnvrtc.so.12'
+        else:
+            _libname = 'nvrtc64_120_0.dll'
+elif CUPY_HIP_VERSION != 0:
+    _prefix = 'hiprtc'
+    if _runtime_version < 5_00_00000:
+        # ROCm 4.x
+        _libname = 'libamdhip64.so.4'
+    elif _runtime_version < 6_00_00000:
+        # ROCm 5.x
+        _libname = 'libamdhip64.so.5'
+
+cdef SoftLink _L = SoftLink(_libname, _prefix, mandatory=True)
+
+
+ctypedef int nvrtcResult
+ctypedef void* nvrtcProgram
+
+# Aliases for compatibillity with existing CuPy codebase.
+# TODO(kmaehashi): Remove these aliases.
+ctypedef nvrtcProgram Program
+
+ctypedef const char* (*F_nvrtcGetErrorString)(nvrtcResult result) nogil
+cdef F_nvrtcGetErrorString nvrtcGetErrorString = <F_nvrtcGetErrorString>_L.get('GetErrorString')  # NOQA
+
+ctypedef nvrtcResult (*F_nvrtcVersion)(int *major, int *minor) nogil
+cdef F_nvrtcVersion nvrtcVersion = <F_nvrtcVersion>_L.get('Version')
+
+ctypedef nvrtcResult (*F_nvrtcCreateProgram)(
+    nvrtcProgram* prog, const char* src, const char* name, int numHeaders,
+    const char** headers, const char** includeNames) nogil
+cdef F_nvrtcCreateProgram nvrtcCreateProgram = <F_nvrtcCreateProgram>_L.get('CreateProgram')  # NOQA
+
+ctypedef nvrtcResult (*F_nvrtcDestroyProgram)(nvrtcProgram *prog) nogil
+cdef F_nvrtcDestroyProgram nvrtcDestroyProgram = <F_nvrtcDestroyProgram>_L.get('DestroyProgram')  # NOQA
+
+ctypedef nvrtcResult (*F_nvrtcCompileProgram)(
+    nvrtcProgram prog, int numOptions, const char** options) nogil
```

Comment:
`ctypedef`s can be generated by copy-pasting the header file.

[`nvrtc.h`](https://gitlab.com/nvidia/headers/cuda-individual/nvrtc/-/blob/88492df6d46cd684bd34df93b307cdc57c9c4dfe/nvrtc.h#L224-225):
```cpp
nvrtcResult nvrtcCompileProgram(nvrtcProgram prog,
                                int numOptions, const char * const *options);
```